apps/api/src/app/workflows-v2/usecases/create-variables-object/create-variables-object.usecase.ts

```diff
@@ -24,7 +39,35 @@ export class CreateVariablesObject {
     ];
     const showIfVariables = this.extractMailyAttribute(controlValues, MailyAttrsEnum.SHOW_IF_KEY);
 
-    return keysToObject(variables, arrayVariables, showIfVariables);
+    const variablesObject = keysToObject(variables, arrayVariables, showIfVariables);
+    if (isEnhancedDigestEnabled) {
+      return this.ensureEventsArray(variablesObject);
+    }
+
+    return variablesObject;
+  }
+
+  private ensureEventsArray(variablesObject: Record<string, unknown>) {
+    const stepsObject = (variablesObject.steps as Record<string, unknown>) ?? {};
+
+    Object.keys(stepsObject)
+      .filter((stepId) => typeof stepsObject[stepId] === 'object')
+      .forEach((stepId) => {
+        const step = stepsObject[stepId] as Record<string, unknown>;
+        const hasUsedEventCount = !!step.eventCount;
+        const hasUsedEventsLength = !!(
+          step.events &&
+          typeof step.events === 'object' &&
+          !Array.isArray(step.events) &&
+          'length' in step.events
+        );
+        const hasUsedEvents = !!(step.events && typeof step.events === 'string');
+        if (hasUsedEventCount || hasUsedEventsLength || hasUsedEvents) {
+          step.events = [];
```

Comment:
I am a bit confused. Should this be just an empty array or an array with a mock payload? Probably I am missing something.

Reply:
the mocking is the next step after this usecase is called in the preview usecase ;)